src/index.js

```diff
@@ -1,19 +1,38 @@
+// @flow
 'use strict';
-require('please-upgrade-node')(require('./package.json'));
+
+require('please-upgrade-node')(require('../package.json'));
 
 const os = require('os');
 const path = require('path');
 const minimist = require('minimist');
-const createExplorer = require('./lib/createExplorer');
+const createExplorer = require('./createExplorer');
 
 const homedir = os.homedir();
 
-module.exports = function cosmiconfig(moduleName, options) {
+module.exports = function cosmiconfig(
+  moduleName: string,
+  options: {
+    packageProp?: string | false,
+    rc?: string | false,
+    js?: string | false,
+    argv?: string | false,
+    format?: 'json' | 'yaml' | 'js',
+    rcStrictJson?: boolean,
+    rcExtensions?: boolean,
+    stopDir?: string,
+    cache?: boolean,
+    sync?: boolean,
+    transform?: (?Object) => ?Object,
+    configPath?: string,
+  }
+) {
   // Keeping argv parsing here allows to mock `minimist` for different tests.
   // This should not have too much of a negative impact.
   const parsedCliArgs = minimist(process.argv);
 
   options = Object.assign(
+    {},
```

Comment:
nit: This shouldn't be required right?

Reply:
This is required. `Object.assign` is pretty hard for Flow to understand.